include/seqan3/alignment/pairwise/alignment_result.hpp

```diff
@@ -200,6 +209,44 @@ class alignment_result
         return data.alignment;
     }
     //!\}
+
+    //!\cond DEV
+    /*!\brief Returns the score matrix used to compute the alignment.
+     * \return The score matrix.
+     *
+     * \details
+     *
+     * This function is only used for debugging such that performance can be affected significantly when enabling
+     * seqan3::align_cfg::debug.
+     *
+     * \attention This function with fail the compilation, if the alignment was not requested in the alignment
```

Comment:
Inconsistency: In the selector in `augment_if_debug` you make a distinction about whether alignment has been selected only for the trace matrix (and not for the score matrix). And here is stated that alignment is necessary for the score matrix, which is not true for your implementation!